controllers/vault_reconciler.go

```diff
@@ -2,31 +2,77 @@ package controllers
 
 import (
 	"context"
-
+	securityv1 "github.com/openshift/api/security/v1"
+	"github.com/prometheus/common/log"
 	workshopv1 "github.com/stakater/workshop-operator/api/v1"
 	"github.com/stakater/workshop-operator/common/kubernetes"
 	"github.com/stakater/workshop-operator/common/util"
 	"github.com/stakater/workshop-operator/common/vault"
-
-	securityv1 "github.com/openshift/api/security/v1"
-	"github.com/prometheus/common/log"
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+var (
+	extraconfigFromValues = map[string]string{
+		"extraconfig-from-values.hcl": `disable_mlock = true
+ui = true
+
+listener "tcp" {
+	tls_disable = 1
+	address = "[::]:8200"
+	cluster_address = "[::]:8201"
+}
+storage "file" {
+	path = "/vault/data"
+}
+`,
+	}
+
+	VaultServerlabels = map[string]string{
+		"app":                       "vault",
+		"app.kubernetes.io/name":    "vault",
+		"app.kubernetes.io/part-of": "vault",
+		"component":                 "server",
+	}
+	VaultAgentlabels = map[string]string{
+		"app":                       "vault",
+		"app.kubernetes.io/name":    "vault-agent-injector",
+		"app.kubernetes.io/part-of": "vault",
+		"component":                 "webhook",
+	}
+)
+
+const (
+	VAULT_NAMESPACE_NAME             = "vault"
+	VAULT_STATEFULSET_NAME           = "vault"
+	VAULT_SERVICE_NAME               = "vault"
+	VAULT_INTERNAL_SERVICE_NAME      = "vault-internal"
+	VAULT_ROLEBINDING_NAME           = "vault-server-binding"
+	VAULT_ROLEBINDING_ROLE_NAME      = "system:auth-delegator"
+	VAULT_ROLEBINDING_KIND_NAME      = "ClusterRole"
```

Comment:
@MaazMS this is not vault specific! this is even more generic; so, better call it

`KIND_CLUSTER_ROLE`